datalad/support/tests/test_annexrepo.py

```diff
@@ -783,7 +783,12 @@ def test_AnnexRepo_commit(path):
         f.write("some")
     assert_raises(FileNotInRepositoryError, ds.commit, files="untracked")
     # not existing file as well:
-    assert_raises(FileNotInRepositoryError, ds.commit, files="not-existing")
+    try:
+        ds.commit(files="not-existing")
+    except FileNotInRepositoryError:
+        pass   # expected result
+    except CommandError:
+        raise SkipTest("test_AnnexRepo_commit hit known failure (gh-4773)")
```

Comment:
eh... I [said](https://github.com/datalad/datalad/issues/4773#issuecomment-668904124) that I am ok to skip it, but I would say we are quite in a good shape in non-windows for not ignoring known failures:
```
$> git grep @known_fail | grep -v -e _win -e appv | grep -e '/test_'
datalad/core/distributed/tests/test_clone.py:@known_failure
datalad/core/distributed/tests/test_clone.py:@known_failure
datalad/support/tests/test_annexrepo.py:@known_failure_v6  #FIXME
```
(so ideally we just annotate with our decorators even though it could mask other failures)...  if you give up -- let me know, I will try too - we should just get to the bottom of this rabbit hole.

Reply:
> eh... I [said](https://github.com/datalad/datalad/issues/4773#issuecomment-668904124) that I am ok to skip it, but I would say we are quite in a good shape in non-windows for not ignoring known failures:
> [...]
> (so ideally we just annotate with our decorators even though it could mask other failures)...

I don't follow your reasoning.  As you suggest, decorating the test disables the testing in all the other parts of the test (and even other unknown failures in this particular call), so why would we ideally do that rather than something more fine-grained like this PR does?

> if you give up -- let me know, I will try too - we should just get to the bottom of this rabbit hole.

Of course any help from others debugging the issue would be appreciated.  But this PR isn't closing that issue; it's just preventing this failure from wasting people's time on nearly every PR.

Reply:
>  so why would we ideally do that rather than something more fine-grained like this PR does?

`known_failure` is easier to find later on.  Sifting through all the skips, some of which are just valid skips not needing to be addressed, would just result in it to get completely forgotten (if we fail to address it now)

Reply:
> `known_failure` is easier to find later on

Sure, but in my view it hardly seems worth throwing away the useful parts of the test in hopes of someone grepping through later _and_ actually fixing it.   If you think it'd help, you could even add a commented out "@known_failure" or otherwise mark it.

Reply:
> you could even add a commented out "@known_failure" or otherwise mark it.

good idea!